tinygrad/runtime/support/hcq.py

```diff
@@ -1,9 +1,28 @@
 from __future__ import annotations
 from typing import List, Optional, Dict, Tuple, cast, Protocol, Type, Union, TypeVar, Generic, Callable, ParamSpec, Concatenate
-import contextlib, decimal, statistics, random, json, atexit, time, array, ctypes, functools
+import contextlib, decimal, statistics, random, json, atexit, time, array, ctypes, functools, os, fcntl
 from tinygrad.helpers import PROFILEPATH, PROFILE, from_mv, getenv
 from tinygrad.renderer import Renderer
 from tinygrad.device import BufferSpec, Compiler, Compiled, LRUAllocator
+from tinygrad.runtime.autogen import libc
+
+# all HCQ interaction with the system happens through this Hardware Abstraction Layer. the devices should not make syscalls
+class HAL:
```

Comment:
idea: HWInterface or something similar would be easier to understand

Reply:
Would love to, but my python skills are non existent 😅 I am mostly watching you guys from the sidelines.